motionplan/armplanning/api.go

```diff
@@ -228,24 +237,35 @@ func PlanMotion(ctx context.Context, logger logging.Logger, request *PlanRequest
 	// goal configurations. However, the blocker here is the lack of a "known good" configuration used to determine which obstacles
 	// are allowed to collide with one another.
 	if request.StartState.configuration == nil {
-		return nil, errors.New("must populate start state configuration")
+		return nil, PlanMeta{}, errors.New("must populate start state configuration")
 	}
 
 	sfPlanner, err := newPlanManager(logger, request)
 	if err != nil {
-		return nil, err
+		return nil, PlanMeta{}, err
 	}
 
-	traj, err := sfPlanner.planMultiWaypoint(ctx)
+	meta := PlanMeta{}
+
+	traj, goalsProcessed, err := sfPlanner.planMultiWaypoint(ctx)
 	if err != nil {
 		if request.PlannerOptions.ReturnPartialPlan {
+			meta.Partial = true
 			logger.Infof("returning partial plan")
 		} else {
-			return nil, err
+			return nil, meta, err
 		}
 	}
 
-	return motionplan.NewSimplePlanFromTrajectory(traj, request.FrameSystem)
+	meta.Duration = time.Since(start)
+	meta.GoalsProcessed = goalsProcessed
+
+	t, err := motionplan.NewSimplePlanFromTrajectory(traj, request.FrameSystem)
```

Comment:
under what conditions will this fail?

if this fails, are we sure it is correct for `meta` to have partial set to true?